### PR TITLE
Translation: "Error Boundaries"

### PR DESCRIPTION
**Pull request by rwachtler**

### CODE REVIEW

File: content/docs/error-boundaries.md [hunk 1]
```diff
@@ -1,28 +1,28 @@
 ---
 id: error-boundaries
-title: Error Boundaries
+title: Fehlergrenzen
 permalink: docs/error-boundaries.html
 ---
 
-In the past, JavaScript errors inside components used to corrupt React’s internal state and cause it to [emit](https://github.com/facebook/react/issues/4026) [cryptic](https://github.com/facebook/react/issues/6895) [errors](https://github.com/facebook/react/issues/8579) on next renders. These errors were always caused by an earlier error in the application code, but React did not provide a way to handle them gracefully in components, and could not recover from them.
+In der Vergangenheit, führten JavaScript-Fehler in Komponenten zu einem fehlerhaften Zustand innerhalb von React, welcher ein [Auftreten](https://github.com/facebook/react/issues/4026) [kryptischer](https://github.com/facebook/react/issues/6895) [Fehlermeldungen](https://github.com/facebook/react/issues/8579) in den folgenden Render-Vorgängen verursachte. Diese Fehler wurden immer von einem zuvor aufgetretenen Fehler in der Applikation verursacht, jedoch war seitens React keine Möglichkeit bereitgestellt, um auf diese Fehler innerhalb der Komponenten zu reagieren und sich von diesen zu erholen.
```
Comment (rwachtler) [inline content/docs/error-boundaries.md]: Should I rather use "Anwendung" instead of "Applikation"? (Consider this for all occurrences of "Applikation")

Comment (ph1p) [reply content/docs/error-boundaries.md]: Use "Anwendung". I think that's a better translation than "Applikation".

Comment (oshell) [inline content/docs/error-boundaries.md]: "um mit Fehlern in Komponenten elegant umzugehen"

File: content/docs/error-boundaries.md [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
 
 
-## Introducing Error Boundaries {#introducing-error-boundaries}
+## Einführung zu Fehlergrenzen {#introducing-error-boundaries}
 
-A JavaScript error in a part of the UI shouldn’t break the whole app. To solve this problem for React users, React 16 introduces a new concept of an “error boundary”.
+Ein JavaScript-Fehler in einem Teil der UI sollte nicht die ganze Applikation zerstören. Um eine Lösung für dieses Problem bereitzustellen, wurde mit React 16 das Konzept einer "Fehlergrenze" vorgestellt.
 
-Error boundaries are React components that **catch JavaScript errors anywhere in their child component tree, log those errors, and display a fallback UI** instead of the component tree that crashed. Error boundaries catch errors during rendering, in lifecycle methods, and in constructors of the whole tree below them.
+Fehlergrenzen sind React-Komponenten die **JavaScript-Fehler im ihrem ganzen Kind-Komponenten Baum abfangen, diese loggen und eine Fallback-UI**, anstatt eines zerstörten Komponenten-Baums anzeigen. Fehlergrenzen fangen Fehler während des Renderings, in Lifecycle-Methoden und in Konstruktoren des ganzen Komponenten-Baums darunter.
```
Comment (oshell) [inline content/docs/error-boundaries.md]: in ihrem ganzen

File: content/docs/error-boundaries.md [hunk 3]
```diff
@@ -32,81 +32,81 @@ class ErrorBoundary extends React.Component {
...
 
-This change has an important implication. **As of React 16, errors that were not caught by any error boundary will result in unmounting of the whole React component tree.**
+Diese Änderung hat eine wesentliche Auswirkung. **Mit React 16 führen Fehler, die nicht von einer Fehlergrenze abgefangen wurden, zum Unmounten des gesamten React Komponenten-Baums.**
 
-We debated this decision, but in our experience it is worse to leave corrupted UI in place than to completely remove it. For example, in a product like Messenger leaving the broken UI visible could lead to somebody sending a message to the wrong person. Similarly, it is worse for a payments app to display a wrong amount than to render nothing.
+Wir haben über diese Entscheidung diskutiert, unserer Erfahrung nach ist es jedoch schlimmer eine fehlerhafte UI anzuzeigen, anstatt diese komplett zu entfernen. Zum Beispiel in einem Produkt wie ein Messenger, wäre es möglich, dass durch das Anzeigen einer fehlerhaften UI, die Nachricht an eine falsche Person verschickt werden könnte. Ähnlich schlimm wäre es in einer Bezahlanwendung, einen falschen Betrag, statt einfach nichts zu rendern.
```
Comment (ph1p) [inline content/docs/error-boundaries.md]:
```suggestion
Wir haben über diese Entscheidung diskutiert, unserer Erfahrung nach ist es jedoch schlimmer eine fehlerhafte UI anzuzeigen, anstatt diese komplett zu entfernen. Zum Beispiel in einem Produkt wie einem Messenger, wäre es möglich, dass durch das Anzeigen einer fehlerhaften UI, die Nachricht an eine falsche Person verschickt werden könnte. Ebenso ist es in einer Zahlunganwendung schlimmer einen falschen Betrag anzuzeigen, statt einfach nichts zu rendern.
```

File: content/docs/error-boundaries.md [hunk 4]
```diff
@@ -116,21 +116,21 @@ try {
...
+Fehlergrenzen bewahren die deklarative Natur von React und verhalten sich so wie du es erwarten würdest. Zum Beispiel, sogar wenn ein Fehler in einer `componentDidUpdate` Methode auftritt, verursacht von einem `setState` irgendwo tief im Komponenten-Baum, wird dieser Fehler trotzdem zur nächstgelegenen Fehlergrenze weitergeleitet.
 
-## How About Event Handlers? {#how-about-event-handlers}
+## Wie sieht es mit Event-Handler aus? {#how-about-event-handlers}
 
-Error boundaries **do not** catch errors inside event handlers.
+Fehlergrenzen fangen **keine** Fehler innerhalb der Event-Handler ab.
 
-React doesn't need error boundaries to recover from errors in event handlers. Unlike the render method and lifecycle methods, the event handlers don't happen during rendering. So if they throw, React still knows what to display on the screen.
+React braucht keine Fehlergrenzen um sich von Fehlern innerhalb von Event-Handler zu erholen. Im Gegensatz zu Render- und Lifecycle-Methoden finden Event-Handler nicht während des Renderings statt. Das heißt, im Falle eines Fehlers weiß React noch immer, was angezeigt werden soll.
```
Comment (ph1p) [inline content/docs/error-boundaries.md]:
```suggestion
React braucht keine Fehlergrenzen um sich von Fehlern innerhalb von Event-Handlern zu erholen. Im Gegensatz zu Render- und Lifecycle-Methoden finden Event-Handler nicht während des Renderings statt. Das heißt, im Falle eines Fehlers weiß React noch immer, was angezeigt werden soll.
```

File: content/docs/error-boundaries.md [hunk 5]
```diff
@@ -116,21 +116,21 @@ try {
...
 
-If you need to catch an error inside event handler, use the regular JavaScript `try` / `catch` statement:
+Wenn du einen Fehler innerhalb von Event-Handler abfangen möchtest, nutze das normale `try` / `catch` JavaScript-Statement:
```
Comment (ph1p) [inline content/docs/error-boundaries.md]:
```suggestion
Wenn du einen Fehler innerhalb von Event-Handlern abfangen möchtest, nutze das normale `try` / `catch` JavaScript-Statement:
```

**Comment by ph1p** [comment]
Thank you (: Well done!

**Pull request merged**